File: jenkins-plugin/src/main/java/org/eazyportal/plugin/release/jenkins/action/SetSnapshotVersionActionFactory.java
package org.eazyportal.plugin.release.jenkins.action;

import com.google.inject.Inject;
import hudson.Extension;
import hudson.model.InvisibleAction;
import org.eazyportal.plugin.release.core.action.SetSnapshotVersionAction;
import org.eazyportal.plugin.release.core.scm.ScmActions;
import org.eazyportal.plugin.release.core.version.SnapshotVersionProvider;
import org.eazyportal.plugin.release.jenkins.ReleaseStepConfig;

import java.io.Serializable;

@Extension
public class SetSnapshotVersionActionFactory extends InvisibleAction implements Serializable {

    @Inject
    private transient ReleaseStepConfig releaseStepConfig;
    @Inject
    private transient SnapshotVersionProvider snapshotVersionProvider;

    public SetSnapshotVersionAction create(ScmActions scmActions) {
        return new SetSnapshotVersionAction(
            scmActions,
            releaseStepConfig.getScmConfig(),
            snapshotVersionProvider
        );
    }

}

File: gradle-plugin/src/main/kotlin/org/eazyportal/plugin/release/gradle/action/SetSnapshotVersionActionFactory.kt
package org.eazyportal.plugin.release.gradle.action

import org.eazyportal.plugin.release.core.action.SetSnapshotVersionAction
import org.eazyportal.plugin.release.core.version.SnapshotVersionProvider
import org.eazyportal.plugin.release.gradle.model.EazyReleasePluginExtension

class SetSnapshotVersionActionFactory : ReleaseActionFactory<SetSnapshotVersionAction> {

    override fun create(extension: EazyReleasePluginExtension): SetSnapshotVersionAction =
        SetSnapshotVersionAction(
            extension.projectActionsFactory,
            extension.scmActions,
            extension.scmConfig,
            SnapshotVersionProvider()
        )

}